mobile/src/full/java/org/openhab/habdroid/core/GcmRegistrationService.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2010-2018, openHAB.org and others.
+ *
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   which accompanies this distribution, and is available at
+ *   http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.habdroid.core;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.google.android.gms.gcm.GoogleCloudMessaging;
+import com.google.android.gms.iid.InstanceID;
+
+import org.openhab.habdroid.core.connection.CloudConnection;
+import org.openhab.habdroid.core.connection.Connection;
+import org.openhab.habdroid.core.connection.ConnectionFactory;
+import org.openhab.habdroid.util.MyHttpClient;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.Locale;
+
+import okhttp3.Call;
+import okhttp3.Headers;
+
+public class GcmRegistrationService extends IntentService {
+    private static final String TAG = GcmRegistrationService.class.getSimpleName();
+
+    static final String ACTION_REGISTER = "org.openhab.habdroid.action.REGISTER_GCM";
+    static final String ACTION_HIDE_NOTIFICATION = "org.openhab.habdroid.action.HIDE_NOTIFICATION";
+    static final String EXTRA_NOTIFICATION_ID = "notificationId";
```

Comment:
Shouldn't the the static from GcmListenerService be used instead of this one? Or the other way around, at least I would recommend not to declare the same constant twice :)

Reply:
I purposefully did that, because I want to decouple the GCM logic (which is only present in full flavor) from the generic activity logic. That's why I defined one constant for each intent 'target'.